devito/builtins/arithmetic.py

```diff
@@ -39,17 +39,12 @@ def norm(f, order=2):
     if f.is_TimeFunction and f._time_buffering:
         kwargs[f.time_dim.max_name] = f._time_size - 1
 
-    # Protect SparseFunctions from accessing duplicated (out-of-domain) data,
-    # otherwise we would eventually be summing more than expected
-    p, eqns = f.guard() if f.is_SparseFunction else (f, [])
```

Comment:
@FabioLuporini so I guess this is up for discussion but I think this is wrong. If someone wants to compute the norm/sum/inner/... of a sparse function, then all you comput is `sum_{t, p} rec[t, p]` so whether or not it lies within the grid is not relevant, The guard should only be used if the physical location `u[posz,...]`  in an interpolation/injection to make sure that index exists in the function.

Reply:
Oh wait no, MPI halo with duplicates right. reverting.